### ISSUE TITLE
Midpoint for semi-infinite interval

### ISSUE THREAD

**Issue by dpsanders** [JuliaIntervals/IntervalArithmetic.jl#107]
[Rump](http://www.ti3.tu-harburg.de/paper/rump/Ru18.pdf) suggests that the midpoint of a semi-infinite interval $[a, \infty]$ should be `sqrt(realmax(T))`

**Comment by lbenet** [comment]
I think our implementation is consistent with the standard, but we should check this up...

**Comment by dpsanders** [comment]
Ah, good point. Why is it so hard to find those things...

**Comment by dpsanders** [comment]
Actually the standard says "no value, if x is empty or unbounded" (page 11) :/

**Comment by oheim** [comment]
The standard also says:

- mid (Empty) = NaN
- mid (Entire) = 0
- mid ([-Infinity, Finite value]) = the finite negative number of largest magnitude
- mid ([Finite value, +Infinity]) = the finite positive number of largest magnitude
- mid (common interval) = the Level 1 value rounded to nearest

It is hard to read implementation stuff from the standard, because the important implementation is split between the Level 1 and 2 descriptions. I'd suggest that you primarily use the (recently published) simplified standard IEEE Std 1788.1.

**Comment by lucaferranti** [comment]
this seems fixed
```julia
julia> mid(emptyinterval())
NaN

julia> mid(entireinterval())
0.0

julia> mid(-Inf..3) == nextfloat(-Inf)
true

julia> mid(1..Inf) == prevfloat(Inf)
true
```
